test/Microsoft_Tests/src/SQLServer_Spec.enso

```diff
@@ -137,12 +126,24 @@ add_specs suite_builder =
                     in_memory.at "txt-limited" . value_type . should_equal (Value_Type.Char size=10 variable_length=True)
                     in_memory.at "txt-fixed" . value_type . should_equal (Value_Type.Char size=3 variable_length=False)
 
-
-                group_builder.specify "basic query" <|
-                    conn = Database.connect get_configured_connection_details
-                    conn.should_succeed
-                    new_database = conn.read (SQL_Query.Raw_SQL "SELECT 'master' as c") . at 0 . to_vector . first
-                    new_database.should_equal "master"
+                group_builder.specify "test datetime2 precision round trip" <|
+                    name = Name_Generator.random_name "datetime2-test"
+                    Problems.assume_no_problems <|
+                        data.connection.execute_update 'CREATE TABLE "'+name+'" ("dt2" DATETIME2)'
+                    t = data.connection.query (SQL_Query.Table_Name name)
+                    row1 = [Date_Time.new 2021 1 1 12 13 14 500 1 1]
+                    row2 = [Date_Time.new 2021 1 1 9 12 12 987 654 321]
+                    row3 = [Nothing]
+                    source_table = Table.from_rows ["dt2"] [row1, row2, row3]
+                    t.update_rows source_table update_action=Update_Action.Insert
+                    ## SQLServer only supports precision to 100 nanoseconds
+                    expected_row1 = [Date_Time.new 2021 1 1 12 13 14 500 1 0]
+                    expected_row2 = [Date_Time.new 2021 1 1 9 12 12 987 654 300]
+                    expected_row3 = [Nothing]
+                    expected_table = Table.from_rows ["dt2"] [expected_row1, expected_row2, expected_row3]
+                    returned_table = t.read
+                    returned_table.should_equal expected_table
+                    data.connection.execute_update 'DROP TABLE "'+name+'"'
```

Comment:
This will not execute if the test fails, leaving trash in the table.

While the CI runs it as docker image, some users may use a persistent DB - we should generally try to leave a clean env afterwards.

Can we just make this a temporary table?? If not, I'd do the DROP inside of `Panic.with_finalizer`

Reply:
I'll make it a temporary after SQL Server has support for them which will be my next ticket. Had trouble doing it manually.